backend/persistent/PlotlyAPI.js

```diff
@@ -36,16 +36,20 @@ export function newDatacache(payloadJSON, type, requestor) {
     const user = users.find(
         u => u.username === requestor
     );
-    const apiKey = user.apiKey;
-    const accessToken = user.accessToken;
 
     let authorization = '';
-    if (apiKey) {
-        authorization = 'Basic ' + new Buffer(
-            requestor + ':' + apiKey
-        ).toString('base64');
-    } else if (accessToken) {
-        authorization = `Bearer ${accessToken}`;
+
+    if (user) {
+        const apiKey = user.apiKey;
+        const accessToken = user.accessToken;
+
+        if (apiKey) {
+            authorization = 'Basic ' + new Buffer(
+                requestor + ':' + apiKey
+            ).toString('base64');
+        } else if (accessToken) {
+            authorization = `Bearer ${accessToken}`;
+        }
```

Comment:
could we add some comments here? reading this code it seems like api key and access token is optional, as we still proceed even if `user` isn't available.

Reply:
Yeah, this is optional. Because for non-private mode or electron app, this will proceed with blank `Authorization` header, but for private mode, User *has* to authenticate, so we will always have atleast accessToken available.

Reply:
I can add a comment mentioning the same ⬆️